crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -55,40 +56,35 @@ async fn check_gas<const A: bool, const B: bool, S>(
     store: &SuiDataStore<A, B, S>,
     gas_payment_id: ObjectID,
     gas_budget: u64,
-) -> SuiResult<(Object, SuiGasStatus<'static>)>
+    is_system_tx: bool,
+) -> SuiResult<SuiGasStatus<'static>>
 where
     S: Eq + Serialize + for<'de> Deserialize<'de>,
 {
-    let gas_object = store.get_object(&gas_payment_id)?;
-    let gas_object = gas_object.ok_or(SuiError::ObjectNotFound {
-        object_id: gas_payment_id,
-    })?;
-    gas::check_gas_balance(&gas_object, gas_budget)?;
-    // TODO: Pass in real computation gas unit price and storage gas unit price.
-    let gas_status = gas::start_gas_metering(gas_budget, 1, 1)?;
-    Ok((gas_object, gas_status))
+    if is_system_tx {
+        Ok(SuiGasStatus::new_unmetered())
```

Comment:
Is the `unmetered` a new way to flag objects?

Reply:
Not sure what "flag" mean here, unmetered means we will no charge gas in the process